tracer/src/Datadog.Trace/Telemetry/Collectors/DependencyTelemetryCollector.cs

```diff
@@ -106,6 +119,20 @@ static bool IsBase32Char(char c)
             }
         }
 
+        private static bool IsGuid(string assemblyName)
+        {
+            switch (assemblyName.Length)
+            {
+                // Simple implementation to remove guids
+                case 36 when assemblyName[8] == '-' && assemblyName[13] == '-' && assemblyName[18] == '-' && assemblyName[23] == '-':
+                // and other weird use cases like ℛ*710fa04a-6428-4dd1-85a0-0419c142709b#5-0 where the suffix can be up to 7 chars long
+                case >= 42 and <= 45 when assemblyName[10] == '-' && assemblyName[15] == '-' && assemblyName[20] == '-' && assemblyName[25] == '-':
```

Comment:
"up to 7 chars" seems optimistic 😉  personally I would just remove the upper bound, but we can always bump it up later